data/creaturescripts/scripts/droploot.lua

```diff
@@ -23,8 +24,9 @@ function onDeath(player, corpse, killer, mostDamage, unjustified, mostDamage_unj
 	else
 		for i = CONST_SLOT_HEAD, CONST_SLOT_AMMO do
 			local item = player:getSlotItem(i)
+			local lossPercent = player:getLossPercent()
 			if item then
-				if table.contains({SKULL_RED, SKULL_BLACK}, player:getSkull()) or math.random(item:isContainer() and 100 or 1000) <= player:getLossPercent() then
+				if isRedOrBlack or math.random(item:isContainer() and 100 or 1000) < lossPercent then
```

Comment:
This is still off by one, I'm afraid.
Imagine it's a container, your random number is 100 and your lossPercent is 100.
You don't lose an item but you should.

I think proper algorithm would be like this:
```suggestion
				if isRedOrBlack or math.random(item:isContainer() and 100 or 1000) <= lossPercent then
```
and also wrap the inner code in `if lossPercet ~= 0 then`...